templates/install.j2

```diff
@@ -71,40 +59,117 @@ ynh_setup_source --dest_dir="$install_dir"
+{% endif %}
+chmod 400 "$install_dir/{{ data.custom_config_file }}"
+chown "$app:$app" "$install_dir/{{ data.custom_config_file }}"
+
+{% if data.generator_mode == 'tutorial' -%}
+### For more complex cases where you want to replace stuff using regexes,
+### you shoud rely on ynh_replace_string (which is basically a wrapper for sed)
+### When doing so, you also need to manually call ynh_store_file_checksum
+###
+### ynh_replace_string --match_string="match_string" --replace_string="replace_string" --target_file="$install_dir/{{ data.custom_config_file }}"
+### ynh_store_file_checksum --file="$install_dir/{{ data.custom_config_file }}"
+{% endif %}
+
 #=================================================
 # SYSTEM CONFIGURATION
 #=================================================
-ynh_script_progression --message="Adding system configurations related to $app..."
+ynh_script_progression "Adding system configurations related to $app..."
+
+{% if data.generator_mode == 'tutorial' -%}
+### `ynh_config_add_phpfpm` is used to set up a PHP config.
+### You can remove it if your app doesn't use PHP.
+### `ynh_config_add_phpfpm` will use the files conf/extra_php-fpm.conf
+### If you're not using these lines:
+###   - You can remove these files in conf/.
+###   - Remove the section "BACKUP THE PHP-FPM CONFIGURATION" in the backup script
+###   - Remove also the section "REMOVE PHP-FPM CONFIGURATION" in the remove script
+###   - As well as the section "RESTORE THE PHP-FPM CONFIGURATION" in the restore script
+###     with the reload at the end of the script.
+###   - And the section "PHP-FPM CONFIGURATION" in the upgrade script
+{% endif %}
 
 {% if data.main_technology == "php" -%}
-# Create a dedicated PHP-FPM config
-# conf/extra_php-fpm.conf will be appended to the auto-generated config, which will go in /etc/php/X.Y/fpm/pool.d/
-ynh_add_fpm_config
+ynh_config_add_phpfpm
 {% endif %}
 
-# Create a dedicated NGINX config using the conf/nginx.conf template
-ynh_add_nginx_config
+ynh_config_add_nginx
 
 {% if data.main_technology not in ["php", "none"] -%}
 {% if data.generator_mode == 'tutorial' -%}
-### `ynh_systemd_config` is used to configure a systemd script for an app, using the conf/systemd.service template
-{% endif %}
-# Create a dedicated systemd config
-ynh_add_systemd_config
+### `ynh_config_add_systemd` is used to configure a systemd script for an app.
+### It can be used for apps that use sysvinit (with adaptation) or systemd.
+### Have a look at the app to be sure this app needs a systemd script.
+### `ynh_config_add_systemd` will use the file conf/systemd.service
+### If you're not using these lines:
+###   - You can remove those files in conf/.
+###   - Remove the section "BACKUP SYSTEMD" in the backup script
+###   - Remove also the section "STOP AND REMOVE SERVICE" in the remove script
+###   - As well as the section "RESTORE SYSTEMD" in the restore script
+###   - And the section "SETUP SYSTEMD" in the upgrade script
```

Comment:
Same stuff as for PHP : this is not relevant because it's automatically handled by the generator